### ISSUE TITLE
PyDeepCL: NetLearner not callable with new trainer types

### ISSUE THREAD

**Issue by mbPDWTeSe33** [hughperkins/DeepCL#50]
```PyDeepCL.NetLearner(adadelta, net, N, images, labels, testN, testImages, testLabels, batchSize)```
results in
```TypeError: Argument 'sgd' has incorrect type (expected PyDeepCL.SGD, got PyDeepCL.Adadelta)```

Same error with Anneal, Nesterov, Adagrad and Rmsprop.

Doc from NeuralNetAPI gives trainer:
```c++
NetLearner netLearner(
    trainer, net,   // <<
    Ntrain, trainData, trainLabels,
    Ntest, testData, testLabels );
```

**Comment by hughperkins** [comment]
Hmmm, right.  I need to check this point.

**Comment by hughperkins** [comment]
Addressed in 2f6dfc9
